### PR TITLE
added tutorial for qr-code-terminal plugin

### PR DESCRIPTION
**Pull request by shwetalsoni**
Added tutorial for `QR code terminal plugin`. 

Screenshot from 2021-06-27 13-33-11
Screenshot from 2021-06-27 13-33-40
Screenshot from 2021-06-27 13-33-56
Screenshot from 2021-06-27 13-34-09
Screenshot from 2021-06-27 13-34-23
Screenshot from 2021-06-27 13-34-35

### CODE REVIEW

**Comment by shwetalsoni** [comment]
Tag #891

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+## 1. Add dependency
+
+For using any plugin, you have to add `wechaty-plugin-contrib` NPM package to the dependencies using the following command:
+
+```ts
+npm i wechaty-plugin-contrib
+```
+
+## 2. Integrating the plugin
```
Comment (proudofsimin) [inline docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: remove the v-ing form. Use imperative sentence.

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+Now, before starting the bot, you can use this plugin:
+
+```ts
+bot.use(QRCodeTerminal(config))
+bot.start()
+```
+
+## 3. Running the bot
```
Comment (proudofsimin) [inline docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: remove the v-ing form. Use imperative sentence.

**Review by proudofsimin — CHANGES_REQUESTED** [review]
Hi, could you explain more about this plugin at the beginning? Is it the users (developers) or customers will scan the QR code? What will the developers receive when someone scan the QR code? And what will it show on phone after scanning the QR code?

**Comment by shwetalsoni** [comment]
> Hi, could you explain more about this plugin at the beginning? Is it the users (developers) or customers will scan the QR code? What will the developers receive when someone scan the QR code? And what will it show on phone after scanning the QR code?

Hii simin,
Actually this plugin is just for demonstrating that how a user can use this plugin in their bot. So, now on scanning this qr code it wont do anything.

I think for better explanation I can provide a conclusion at end reffering to ding dong bot because there we have used this plugin. On scanning that qr code users can connect to whatsapp and wechat.

What do you think or is their any better way to do this ?

**Review by proudofsimin — COMMENTED** [review]
Hi @shraddhavp,

Let me clear up. So in here, you use the plugin to generate a QR code first, and then you are required to use the events (are they called events?) to define the function for it. Am I right? I agree with your suggestion that mentioning a real example really helps me/user to clear it up. Thank you.

**Comment by shwetalsoni** [comment]
Hii @proudofsimin
 I have changed ing(s) to imperative form. 
I have also added conclusion section at end describing about the application of `QR code terminal` plugin. 
I have updated the screenshots for better understanding.
Thank you :)

**Comment by shwetalsoni** [comment]
> Hi @shraddhavp,
> 
> Let me clear up. So in here, you use the plugin to generate a QR code first, and then you are required to use the events (are they called events?) to define the function for it. Am I right? I agree with your suggestion that mentioning a real example really helps me/user to clear it up. Thank you.

yeah, you get it right. I have made the changes now. You can review it. 
Btw it's not @shraddhavp :sweat_smile:

**Comment by proudofsimin** [comment]
> Hii @proudofsimin
> I have changed ing(s) to imperative form.
> I have also added conclusion section at end describing about the application of `QR code terminal` plugin.
> I have updated the screenshots for better understanding.
> Thank you :)

Thanks for the modification and adding the starter bot. Could you put the "create the starter bot" in step one like #963 ? Thank you.

P.s. sorry for linking the wrong name...

**Review by proudofsimin — APPROVED** [review]
Since it's only minor modification left, approve first.

**Comment by shwetalsoni** [comment]
> > Hii @proudofsimin
> > I have changed ing(s) to imperative form.
> > I have also added conclusion section at end describing about the application of `QR code terminal` plugin.
> > I have updated the screenshots for better understanding.
> > Thank you :)
> 
> Thanks for the modification and adding the starter bot. Could you put the "create the starter bot" in step one like #963 ? Thank you.
> 
> P.s. sorry for linking the wrong name...

I have added this step

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+:::
+
+## Adding QR code terminal plugin
+
+For adding this plugin, refer to **Starter Bot** for building a basic bot and then to add the `QR code terminal` plugin to it follow the steps below:
+
+### 1. Create a starter bot
+
+Follow the instructions on the **Starter Bot** page to create the foundation of a Wechaty bot.
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: I think we should add a starter bot link here for **Starter Bot**

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+```ts
+bot.use(QRCodeTerminal(config))
+bot.start()
+```
+
+### 4. Run the bot
+
+Now, you are done with the integration of `QR code terminal plugin` with your bot. For running the bot you have to **export/set** an environment variable with the type of puppet to use and then run the bot.
+
+#### Linux/macOS
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: I think we should unify the style here, you did a good job at #888 , it looks like this：

image

You can use code like this:

```markdown
<Tabs
  groupId="operating-systems"
  defaultValue="linux"
  values={[
    { label: 'Linux',   value: 'linux', },
    { label: 'macOS',   value: 'mac', },
    { label: 'Windows', value: 'windows', },
  ]
}>

<TabItem value="linux">

```

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+Congratulations! you have successfully added `QR code terminal` plugin to your bot. On running the bot it will show a QR code on terminal.
+
+![QR code terminal plugin output](../../static/img/docs/qrcode-bot/output.png)
+
+## Conclusion
+
+You can use this `QR code terminal` plugin for building the bots where you have to connect to a device. You can refer to [Ding Dong bot](../getting-started/hard-way.mdx). In ding dong bot you have to scan the generated QR code to connect to **Whatsapp/Wechat**.
+
+## References
+
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: Add video link reference here: https://www.youtube.com/watch?v=tfGZXoe_aA4

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+* [GitHub repository of Wechaty Plugin Contrib](https://github.com/wechaty/wechaty-plugin-contrib)
+
```
Comment (lijiarui) [inline docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: Pls remove blank lines here.

**Review by lijiarui — CHANGES_REQUESTED** [review]
Pls change as I request.

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md (hunk 3 above)
Comment (shwetalsoni) [reply docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: Yeah, I will add the link here. Started Bot will be under `examples/basic/`. We have this section in the upcoming weeks in our proposal. So once the starter bot is added and merged we can provide the link here.

**Comment by shwetalsoni** [comment]
@lijiarui I have made the suggested changes except for the starter bot link which I'll add later when the starter bot is created and merged.

File: docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md (hunk 3 above)
Comment (lijiarui) [reply docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md]: I think you could file a new issue for this. Once you create a new issue, I will ignore this part.

**Comment by lijiarui** [comment]
Good job! After you file a new issue for [this](https://github.com/wechaty/wechaty.js.org/pull/969#discussion_r665081568) and link to this PR, I will approve this PR.

**Comment by shwetalsoni** [comment]
@lijiarui I have created issue #1032 for keeping the record of all the features that need to be added later mentioning all the respective PR's.

**Review by lijiarui — APPROVED** [review]
Good job！ Appoved

**Review by huan — APPROVED** [review]

**Pull request merged**